packages/components/src/menu-item/index.js

```diff
@@ -79,7 +84,7 @@ export function MenuItem(
 				className="components-menu-item__shortcut"
 				shortcut={ shortcut }
 			/>
-			{ icon && <Icon icon={ icon } /> }
+			{ icon && iconPosition === 'right' && <Icon icon={ icon } /> }
```

Comment:
[same as comment at https://github.com/WordPress/gutenberg/pull/34710#discussion_r707567899, since this PR includes the same changes]

This way of adding an icon to the right of the Button's content (as opposed to the left) feels hacky — have you considered adding this functionality directly on the Button's component?

Reply:
After discussing this a bit more with @ciampo in dm and exploring the code a bit, it seems that we have been hacking our way around for many current controls. It seems that last year [I had added](https://github.com/WordPress/gutenberg/pull/26687/files?file-filters%5B%5D=.js&file-filters%5B%5D=.md#diff-1204d0def76e28ff29950465cc842e3fafcee022c7e97c76fa5cf5baa4c4505aR119) `iconPosition` in Button but used it with `text` [added here](https://github.com/WordPress/gutenberg/pull/26425/files).

[Tool Selector](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/tool-selector/index.js#L77) uses a similar 'hack' by passing the `icon` as a `label` along with text.

So these usages and others that will surface by searching need to be consolidated through a more consistent API. We will probably need to make `MenuItem` always use the `icon` as a prop and also pass the `iconPosition`. This requires a greater effort and being extra cautious not to brake existing UIs and should be addressed in a separate PR.